src/cairoUtilFuncGen/enumInputCheck.ts

```diff
@@ -0,0 +1,84 @@
+import assert from 'assert';
+import {
+  ASTNode,
+  DataLocation,
+  EnumDefinition,
+  Expression,
+  FunctionCall,
+  FunctionStateMutability,
+  getNodeType,
+  IntType,
+  TypeNode,
+} from 'solc-typed-ast';
+import { FunctionStubKind } from '../ast/cairoNodes';
+import { cloneASTNode } from '../utils/cloning';
+import { createCairoFunctionStub, createCallToFunction } from '../utils/functionGeneration';
+import { typeNameFromTypeNode } from '../utils/utils';
+import { StringIndexedFuncGen } from './base';
+
+export class EnumInputCheck extends StringIndexedFuncGen {
+  gen(
+    node: Expression,
+    nodeInput: Expression,
+    enumDef: EnumDefinition,
+    nodeInSourceUnit: ASTNode,
+  ): FunctionCall {
+    const nodeType = getNodeType(node, this.ast.compilerVersion);
+    const inputType = getNodeType(nodeInput, this.ast.compilerVersion);
+
+    this.sourceUnit = this.ast.getContainingRoot(nodeInSourceUnit);
+    const name = this.getOrCreate(inputType, enumDef);
+    const functionStub = createCairoFunctionStub(
+      name,
+      [['arg', typeNameFromTypeNode(inputType, this.ast), DataLocation.Default]],
+      [['ret', typeNameFromTypeNode(nodeType, this.ast), DataLocation.Storage]],
+      ['range_check_ptr'],
+      this.ast,
+      nodeInSourceUnit ?? nodeInput,
+      FunctionStateMutability.Pure,
+      FunctionStubKind.FunctionDefStub,
+    );
+    const functionInput = cloneASTNode(nodeInput, this.ast);
+
+    this.ast.setContextRecursive(functionInput);
+    return createCallToFunction(functionStub, [functionInput], this.ast);
+  }
+
+  private getOrCreate(type: TypeNode, enumDef: EnumDefinition): string {
+    const key = type.pp();
```

Comment:
key needs to be more specific than this. Needs to include both the enum and the input type